File: Domain/Entities/Alert.cs
namespace Domain.Entities;

public class Alert
{
    public int Id { get; set; }

    public int? ProductId { get; set; }
    public Product? Product { get; set; }

    public string AlertType { get; set; } = null!;
    public string Severity { get; set; } = "warning";
    public string Title { get; set; } = null!;
    public string Description { get; set; } = null!;

    public bool IsAcknowledged { get; set; }
    public bool IsResolved { get; set; }

    public DateTime CreatedAtUtc { get; set; } = DateTime.UtcNow;
    public DateTime? AcknowledgedAtUtc { get; set; }
    public DateTime? ResolvedAtUtc { get; set; }
}

File: client/src/types/alert.ts
export type AlertListItem = {
  id: number;
  productId: number | null;
  sku: string;
  productName: string;
  alertType: string;
  severity: string;
  title: string;
  description: string;
  status: string;
  createdAtUtc: string;
  resolvedAtUtc: string | null;
};

export type AlertListResponse = {
  items: AlertListItem[];
  totalCount: number;
  page: number;
  pageSize: number;
  totalPages: number;
  openCount: number;
  acknowledgedCount: number;
  resolvedCount: number;
  criticalCount: number;
  warningCount: number;
  infoCount: number;
};

export type AlertQueryParameters = {
  search?: string;
  status?: string;
  severity?: string;
  alertType?: string;
  page?: number;
  pageSize?: number;
};